types/queries.go

```diff
@@ -123,6 +125,28 @@ type AllBalancesResponse struct {
 	Amount Coins `json:"amount"`
 }
 
+type DenomMetadataQuery struct {
+	Denom string `json:"denom"`
+}
+
+type DenomMetadataResponse struct {
+	Metadata DenomMetadata `json:"metadata"`
+}
+
+type AllDenomMetadataQuery struct {
+	// optional argument
```

Comment:
Nit: Here and others: Godoc comments should start with the attribute name, like: `Pagination is an optional argument.`